thunder/core/transform_common.py

```diff
@@ -174,7 +192,9 @@ def _helper(x):
                 needed_proxies.add(variableify(x))
 
     dcetrace = from_trace(trace)
-    dcetrace.bound_symbols = list(reversed(dced))
+    dced_bound_symbols = list(reversed(dced))
+    dced_bound_symbols = dced_bound_symbols
+    dcetrace.bound_symbols = remove_duplicate_number_proxies(dced_bound_symbols)
```

Comment:
This is so interesting. Would you add a comment here, or maybe to the description of the dce pass itself, about how duplicate number proxies might happen? What does it look like to have them and then to remove them?

fyi @jjsjann123

Reply:
Well, so basically remove the line and run the tests.